public/js/History.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * History: Keep track of user's sample history.
+ *
+ * This object emulates the typical back/forward behaviour in browsers.
+ *
+ * @constructor
+ */
+function History() {
+    this.sampleHistory = [];
+    this.iterator = 0;
+    this.length = this.sampleHistory.length;
+    this.lastSample = null;
+}
+
+/**
+ * add: Adds an item to the history.
+ *
+ * Any elements after the current iterator are removed.
+ *
+ * @this {History}
+ * @param {string} sampleId - The sampleID to add to the history.
+ */
+History.prototype.add = function(sampleId) {
+    if(sampleId !== this.lastSample) {
+        this.sampleHistory = this.sampleHistory.slice(0, this.iterator+1);
+        this.sampleHistory.push(sampleId);
+        this.iterator = this.sampleHistory.length - 1;
+        this.length = this.sampleHistory.length;
+        this.lastSample = sampleId;
+    }
```

Comment:
Shouldn't there be an else? Or is the if clause to prevent no-op clicks from adding to the history? If the latter, please add a comment to indicate this!